apluslms_roman/backends/__init__.py

```diff
@@ -67,21 +67,25 @@ def __str__(self):
 
 
 class BuildResult:
-    __slots__ = ('code', 'error', 'step')
+    __slots__ = ('step', 'ok', 'code', 'error')
 
-    def __init__(self, code=0, error=None, step=None):
-        self.code = code
-        self.error = error
-        self.step = step
-        assert self.ok or step is not None, "step is required for failed result"
+    def __init__(self, step, ok=True, code=0, error=None):
```

Comment:
Step is not very logical for `ok` result, as its not specific step where the build succeeded. But when build is stopeed before the end, we would like to know where it was stopped (and why).